### ISSUE TITLE
security-setup failure

### ISSUE THREAD

**Issue by Shree1916** [CiscoCloud/microservices-infrastructure#1093]
```
Traceback (most recent call last):
  File "./security-setup", line 971, in <module>
    main(parser.parse_args())
  File "./security-setup", line 957, in main
    print(' {} '.format(cls.__name__).center(40, '='))
ValueError: zero length field name in format
```

**Comment by siddharthist** [comment]
Can you please post your python version?

**Comment by Shree1916** [comment]
Python 2.6.6

**Comment by sehqlr** [comment]
I think that the oldest version of Python 2 that I've seen installed anywhere is 2.7.5. I'll see if there is a simple fix.

**Comment by siddharthist** [comment]
@Shree1916 Can you try with a newer version of python 2? Just so we can isolate or reject this as a python version problem.

**Comment by Shree1916** [comment]
Had both 2.6.6 and 2.7.6 version of python. Now made 2.7.6 as the default one. But still getting same error
![error](https://cloud.githubusercontent.com/assets/17061818/12908423/31ec075c-cf1d-11e5-9ef1-605db33dd931.jpg)

**Comment by zoza1982** [comment]
<img width="1177" alt="screen shot 2016-02-08 at 10 01 09 pm" src="https://cloud.githubusercontent.com/assets/11164593/12908847/83a88156-ceaf-11e5-8f68-e3400e036ac2.png">

I have exact version as you and it works fine...on MacOSX Yosemite 10.10.5

**Comment by zoza1982** [comment]
Inside of security-setup script. Try pocking around the first line "#!/usr/bin/env python"  and hard code it to exact path of your python 2.7.6

**Comment by zoza1982** [comment]
1) Try git cloning again.....to a fresh folder. Maybe data got currupted in transfer. Rare case but it cat happend.
2) do "file security.yml" to check for file type  it should say "security-setup: a python script text executable"
3) try "dos2unix security-setup " maybe it picked up some funky characters by being copied from windows machine? It happens sometimes...

**Comment by Shree1916** [comment]
I followed the below:

1. Checked out the code again.
2. Changed the python path in security-setup to 2.7

Now getting new error:
**[root@puppet-master microservices-infrastructure]# ./security-setup 
Traceback (most recent call last):
  File "./security-setup", line 20, in <module>
    import yaml
ImportError: No module named yaml**

**Comment by siddharthist** [comment]
@Shree1916 You'll have to run `pip install -r requirements.txt` before `security-setup` will work. Relevant documentation [here](http://microservices-infrastructure.readthedocs.org/en/latest/getting_started/index.html?highlight=pip).

**Comment by Shree1916** [comment]
Yes, followed that as well.

![error2](https://cloud.githubusercontent.com/assets/17061818/12909547/ee8d0320-cf27-11e5-9084-b761655123e9.jpg)

**Comment by siddharthist** [comment]
@Shree1916 This definitely seems like a problem in your environment. Try creating a virtualenv, I guess? I'm not sure why your python installation can't find what you're installing with pip. In any case, I don't think this is related to Mantl code AFAICT.

**Comment by sehqlr** [comment]
@siddharthist I took at look at the confusing section of code, and I actually understand it alright. I'll see about adding some commentary there. 

@Shree1916 let me know if you are still having issues. It looks like the `__name__` attribute for the class is empty, but I don't know how that could be, since `Component` inherits from `object` and therefore, should get all the new-style class features. I installed 2.6.6 with `pyenv`, and the setup script ran successfully.
